classes/Sensors/System.php

```diff
@@ -266,4 +293,75 @@ public function WPUpdate($automatic)
             ));
         }
     }
+
+    /**
+     * Purge log files older than one month
+     */
+    public function LogFilesPruning()
+    {
+        if ($this->plugin->GetGlobalOption('purge-log', 0)) {
+            $upload_dir = wp_upload_dir();
+            $uploadsDirPath = trailingslashit($upload_dir['basedir']).'404s/';
+            if (is_dir($uploadsDirPath)) {
+                if ($handle = opendir($uploadsDirPath)) {
+                    while (false !== ($entry = readdir($handle))) {
+                        if ($entry != "." && $entry != "..") {
+                            if (file_exists($uploadsDirPath . $entry)) {
+                                $modified = filemtime($uploadsDirPath . $entry);
+                                if ($modified < strtotime('-4 weeks')) {
+                                    // Delete file
+                                    unlink($uploadsDirPath . $entry);
+                                }
+                            }
+                        }
+                    }
+                    closedir($handle);
+                }
+            }
+        }
+    }
+
+    /**
+     * Write a new line on 404 log file
+     */
+    private function WriteLog($data, $ip, $username = '')
+    {
+        $upload_dir = wp_upload_dir();
+        $uploadsDirPath = trailingslashit($upload_dir['basedir']).'404s/';
+        // Check directory
+        if (is_dir($uploadsDirPath) && is_readable($uploadsDirPath) && is_writable($uploadsDirPath)) {
+            $fp = $uploadsDirPath . $ip . $username . '.log';
```

Comment:
@carmelopreeostudios start with 0 immediately so there's no need to open 2 files.